application/config/config.go

```diff
@@ -866,6 +884,28 @@ func (c *Config) IsFedramp() bool {
 	host := strings.ToLower(u.Hostname())
 
 	// fedramp instance should have the format https://*.snykgov.io
-	snykgovDomain := strings.HasSuffix(host, ".snykgov.io")
-	return snykgovDomain
+	return isFedrampHost(host)
+}
+
+func (c *Config) GetSnykCodeApi() (string, error) {
+	if !c.IsFedramp() {
+		return c.SnykCodeApi(), nil
+	}
+
+	// api url is fedramp, but code api url hasn't been updated properly
+	if strings.Contains(c.SnykCodeApi(), "deeproxy.") {
+		snykCodeApiUrl, err := getCodeApiUrlFromCustomEndpoint(c.SnykApi(), c.IsFedramp())
+		if err != nil {
+			return "", err
+		}
+		c.SetSnykCodeApi(snykCodeApiUrl)
```

Comment:
Getter functions shouldn't be mutating. As a caller of `Get` I expect it's not doing any side-effect mutation.